src/screens/PrivateChat/ContactsPicker/index.js

```diff
@@ -0,0 +1,252 @@
+import styles from './styles';
+import FakeNavigationBar from '../../../components/common/FakeNavigationBar';
+import Loading from '../../../components/common/Loading';
+import NavigatorComponent from '../../../components/common/NavigatorComponent';
+import i18n from '../../../global/i18n';
+import Colors from '../../../global/colors';
+import type { Contact } from '../../../types/Contacts';
+import type { Navigator } from '../../../types/ReactNativeNavigation';
+import ScreenTitle from '../../../components/common/ScreenTitle';
+import AssetsImage from '../../../global/AssetsImages';
+import { imageSource } from '../../../utils/profile';
+import ListItem from '../../../components/common/ListItem';
+
+const DONE_BUTTON = 'DONE_BUTTON';
+const DISABLED_RIGHT_BUTTON = {
+  title: i18n.t('screens.contactsPicker.doneButton'),
+  id: DONE_BUTTON,
+  buttonColor: Colors.navigationButtonColor,
+  disabled: true,
+};
+const ADD_CONTACT_ERROR = i18n.t('screens.contactsPicker.addContactError');
+
+type Props = {
+  /**
+   * @desc React Native Navigation navigator object.
+   */
+  navigator: Navigator,
+  /**
+   * @desc List of all contacts.
+   */
+  contacts: Array<Contact>,
+  /**
+   * @desc List of initially selected contacts.
+   */
+  initialSelectedContacts: Array<Contact>,
+  /**
+   * @desc Function to add a new contact
+   * @param {string} identityKey Identity key of user.
+   * @param {function} callback Callback
+   */
+  addContact: (identityKey: string, callback: (error: Error | null) => void) => void,
+  /**
+   * @desc Callback function to call when submitting contact selection.
+   */
+  onContactsSelected: (selectedContacts: Array<Contact>) => void,
+};
+
+type State = {
+  /**
+   * @desc The list of selected contacts.
+   */
+  selectedContacts: Array<Contact>,
+  /**
+   * @desc The identityKey of the added contact.
+   */
+  addedContactIdentityKey: string,
+  /**
+   * @desc Flag whether loading is in progress.
+   */
+  loading: boolean,
+  /**
+   * @desc Error message for adding contact failure.
+   */
+  addContactError: string,
+  /**
+   * @desc Flag whether done button is disabled.
+   */
+  doneBtnDisabled: boolean,
+};
+
+class ContactsPickerScreen extends NavigatorComponent<Props, State> {
+  static navigatorButtons = {
+    leftButtons: [],
+    rightButtons: [DISABLED_RIGHT_BUTTON],
+  };
+
+  static defaultProps = {
+    initialSelectedContacts: [],
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedContacts: [...this.props.initialSelectedContacts],
+      addedContactIdentityKey: '',
+      loading: false,
+      addContactError: '',
+      doneBtnDisabled: true,
+    };
+  }
+
+  selectize: Selectize;
+
+  onNavBarButtonPress(id) {
+    if (id === DONE_BUTTON) {
+      this.props.onContactsSelected(this.state.selectedContacts);
+    }
+  }
+
+  componentDidMount() {
+    const doneButtonDisabled = this.props.initialSelectedContacts.length === 0;
```

Comment:
@migbot Yes, I think we shouldn't allow user to save, unless there is a change. So, then there is no additional code required, right? Since we never want to update on constructor, only on update.